task2.cpp

```diff
@@ -0,0 +1,175 @@
+#include<iostream>
+#include<fstream>
+#pragma warning (disable: 4996)
+using namespace std;
+const char fileName[] = "offers.bin";
+const char a = 'a';
+const char i = 'i';
+const char s = 's';
+const char f = 'f';
+const char q = 'q';
+struct Offer {
+	char firmName[25];
+	int coworkers;
+	int offDays;
+	long long payment;
+};
+
+size_t getFileSize(ifstream& f) {
+	size_t currentPos = f.tellg();
+	f.seekg(0, ios::end);
+	size_t size = f.tellg();
+
+	f.seekg(currentPos);
+	return size+1;
+}
+
+void enterAttributes(char* firmName, int& coworkers, int& offDays, long long& payment) {
+	cout << "What is the name of the firm:";
+	cin >> firmName;
+	cout<<"How many are the coworkers:";
+	cin >> coworkers;
+	cout << "How many are the days off:";
+	cin >> offDays;
+	cout << "How big is the payment";
+	cin >> payment;
+	cout << "-------------------------" << endl;
+}
+
+void createOffer(Offer& offer) {
+	char firmName[25];
+	int collegues,offDays;
+	long long payment;
+	enterAttributes(firmName, collegues, offDays, payment);
+	
+	strcpy(offer.firmName, firmName);
+	offer.coworkers = collegues;
+	offer.offDays = offDays;
+	offer.payment = payment;
+}
+
+void writeOneItemTofile(ofstream& out, Offer* offer) {
+	out.write((const char*)&offer->firmName, sizeof(offer->firmName));
+	out.write((const char*)&offer->coworkers, sizeof(offer->coworkers));
+	out.write((const char*)&offer->offDays, sizeof(offer->offDays));
+	out.write((const char*)&offer->payment, sizeof(offer->payment));
+}
+
+void writeToFileOffers(ofstream& out, Offer* offers, int n) {
+	for (int i = 0; i < n; i++) {
+		writeOneItemTofile(out,&offers[i]);
+	}
+}
+
+void readOneItemFromFile(ifstream& in, Offer* offer) {
```

Comment:
read the whole array

Reply:
you can read all orders with one single call of .read